src/app/embed/embed.component.ts

```diff
@@ -28,27 +28,33 @@ export class EmbedComponent implements OnInit {
   subscribeTarget: string;
   artworkUrl: string;
 
-  constructor(private route: ActivatedRoute) {}
-
-  ngOnInit() {
-    this.route.queryParams.forEach(params => {
-      if (params[EMBED_AUDIO_URL_PARAM]) {
-        this.audioUrl        = params[EMBED_AUDIO_URL_PARAM];
-        this.title           = params[EMBED_TITLE_PARAM];
-        this.subtitle        = params[EMBED_SUBTITLE_PARAM];
-        this.subscribeUrl    = params[EMBED_SUBSCRIBE_URL_PARAM];
-        this.subscribeTarget = params[EMBED_SUBSCRIBE_TARGET];
-        this.artworkUrl      = params[EMBED_IMAGE_URL_PARAM];
-      }
-   });
-  }
-
-  showModal() {
-    this.showShareModal = true;
-  }
-
-  hideModal() {
-    this.showShareModal = false;
-  }
+  constructor(
+		private route: ActivatedRoute,
+		private http: Http
+	) {}
+
+	ngOnInit() {
+		this.route.queryParams.forEach(params => {
+			const adapter = new MergeAdapter(params, this.http)
```

Comment:
I think I would rather have this injected, rather than constructing it here. Also, the string property access is another indicator to me that we should define an interface.